packages/api-server/src/base/gw-config.ts

```diff
@@ -0,0 +1,351 @@
+      validatorCodeHash: sudt.validatorCodeHash,
+      generatorCodeHash: sudt.generatorCodeHash,
+      validatorScriptTypeHash: sudt.validatorScriptTypeHash,
+    },
+    meta: {
+      validatorCodeHash: meta.validatorCodeHash,
+      generatorCodeHash: meta.generatorCodeHash,
+      validatorScriptTypeHash: meta.validatorScriptTypeHash,
+    },
+    polyjuice: {
+      validatorCodeHash: polyjuice.validatorCodeHash,
+      generatorCodeHash: polyjuice.generatorCodeHash,
+      validatorScriptTypeHash: polyjuice.validatorScriptTypeHash,
+    },
+    ethAddrReg: {
+      validatorCodeHash: ethAddrReg.validatorCodeHash,
+      generatorCodeHash: ethAddrReg.generatorCodeHash,
+      validatorScriptTypeHash: ethAddrReg.validatorScriptTypeHash,
+    },
+  };
+  return backends;
+}
+
+export interface ConfigGwScripts {
+  deposit: Omit<GwScript, "scriptType">;
+  withdraw: Omit<GwScript, "scriptType">;
+  stateValidator: Omit<GwScript, "scriptType">;
+  stakeLock: Omit<GwScript, "scriptType">;
+  custodianLock: Omit<GwScript, "scriptType">;
+  challengeLock: Omit<GwScript, "scriptType">;
+  l1Sudt: Omit<GwScript, "scriptType">;
+  l2Sudt: Omit<GwScript, "scriptType">;
+  omniLock: Omit<GwScript, "scriptType">;
+}
+
+export function toConfigGwScripts(nodeInfo: NodeInfo) {
+  const deposit = findGwScript(GwScriptType.Deposit, nodeInfo);
+  const withdraw = findGwScript(GwScriptType.Withdraw, nodeInfo);
+  const stateValidator = findGwScript(GwScriptType.StateValidator, nodeInfo);
+  const stakeLock = findGwScript(GwScriptType.StakeLock, nodeInfo);
+  const custodianLock = findGwScript(GwScriptType.CustodianLock, nodeInfo);
+  const challengeLock = findGwScript(GwScriptType.ChallengeLock, nodeInfo);
+  const l1Sudt = findGwScript(GwScriptType.L1Sudt, nodeInfo);
+  const l2Sudt = findGwScript(GwScriptType.L2Sudt, nodeInfo);
+  const omniLock = findGwScript(GwScriptType.OmniLock, nodeInfo);
+
+  const configGwScripts: ConfigGwScripts = {
+    deposit,
+    withdraw,
+    stateValidator,
+    stakeLock,
+    custodianLock,
+    challengeLock,
+    l1Sudt,
+    l2Sudt,
+    omniLock,
+  };
+  return configGwScripts;
+}
+
+export function findGwScript(type: GwScriptType, nodeInfo: NodeInfo): GwScript {
+  const script = nodeInfo.gwScripts.find((s) => s.scriptType === type);
+  if (script == null) {
+    throw new Error(`[GwConfig => findGwScript] can not find script ${type}`);
+  }
+  return script!;
+}
+
+export interface ConfigEoaScripts {
+  eth: Omit<EoaScript, "eoaType">;
+}
+
+export function toConfigEoaScripts(nodeInfo: NodeInfo) {
+  const eth = nodeInfo.eoaScripts.find((e) => e.eoaType === EoaScriptType.Eth);
+  if (eth == null) {
+    throw new Error("no Eth eoa script!");
+  }
+
+  const configEoas: ConfigEoaScripts = {
+    eth,
+  };
+  return configEoas;
+}
+
+export function toApiNodeInfo(nodeInfo: GwNodeInfo): NodeInfo {
+  return snakeToCamel(nodeInfo, ["code_hash", "hash_type"]);
+}
+
+export async function findFirstEoaAccountId(
```

Comment:
export or not ?
If no need to call outside this file, we can make there functions private.